### PR TITLE
New storage/box base type for crates

### PR DESCRIPTION
**Pull request by Sanuki357**
## About The Pull Request

This changes all the crates that previously used **/obj/item/storage/box** as the base type, now uses **/obj/item/storage/box/crate** which has a altered text when the box is folded and gives a sheet of wood when folded, in which previously dispensed a crumpled paper despite the wooden appearance of the crate.

This also adds a new sprite of a crate for the ST-580 point defense sentry, which previously used the same sprite as the ST-571 sentry gun.
[This also gives a state for when the crate is empty](https://github.com/tgstation/TerraGov-Marine-Corps/pull/13670).

The new sprite is imported into TerraGov-Marine-Corps\icons\Marine\marine-weapons.dmi where the original sentry box is stored, along with the recently added empty crate sprite.

I made this pull because after my last [pull request](https://github.com/tgstation/TerraGov-Marine-Corps/pull/13670), I did not notice that the minisentry crate lacked the empty sprite, and they were using the same sprite as the regular sentry.

https://github.com/tgstation/TerraGov-Marine-Corps/assets/58515178/b2d7b214-b398-4c50-93f2-89777258f0ca
## Why It's Good For The Game
It was odd to see a wooden crate get folded into a crumpled paper as if you are a wise man in China under the rule of Han Dynasty.

The new sprite for the minisentry gives a visual differentiation between the regular sentry and the minisentry.
The new sprite bares the smaller icon on its crate, hopefully giving a impression to players that it is a smaller version of the regular sentry.

As for the empty version of the crate, ditto from my previous [pull](https://github.com/tgstation/TerraGov-Marine-Corps/pull/13670), but for all of the wooden crates featured in TGMC.
## Changelog
:cl:
imageadd: added sprite for the crate of ST-580 point defense sentry which was using the same sprite as the ST-571 sentry gun
imageadd: all the wooden crates (the one you can hold on like the sentry one) now has the sprite when they are empty
code: all the wooden crates now dispenses a wooden plank when folded
/:cl:

### CODE REVIEW

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Just make the base crate do this instead of adding it to crates one at a time, since they can all share an open icon.

**Comment by Sanuki357** [comment]
> Just make the base crate do this instead of adding it to crates one at a time, since they can all share an open icon.

Ok that'll involve a bit more extensive changes then, since all the wooden crates uses /obj/item/storage/box which has many others uses than a wooden crates.
I guess what I should do is define a new /obj/item/storage for crates?

**Comment by Lumipharon** [comment]
ye

**Comment by Sanuki357** [comment]
> ye

sir yes sir oorah

File: code/game/objects/items/storage/marine_boxes.dm [hunk 1]
```diff
@@ -1,47 +1,78 @@
...
+		return
+
+	// Close any open UI windows first
+	for(var/mob/M in content_watchers)
+		close(M)
+
+	// Now make the plank
+	to_chat(user, span_notice("You broken [src] into a plank."))
+	new foldable(get_turf(src))
+	qdel(src)
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:29]: why does this exist?

File: code/game/objects/items/storage/marine_boxes.dm [hunk 2]
```diff
@@ -1,47 +1,78 @@
...
+
+/obj/item/storage/box/crate/update_icon_state()
+	icon_state = initial(icon_state)
+	if(!length(contents))
+		icon_state = "empty_case"
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:34]: Write this as a ternary operator or something, there's no reason to be changing the icon state twice for no reason.

File: code/game/objects/items/storage/marine_boxes.dm [hunk 3]
```diff
@@ -1,47 +1,78 @@
...
+/obj/item/storage/box/crate/heavy_armor
 	name = "\improper B-Series defensive armor crate"
 	desc = "A large case containing an experiemental suit of B18 armor for the discerning specialist."
-	icon = 'icons/Marine/marine-weapons.dmi'
 	icon_state = "armor_case"
 	w_class = WEIGHT_CLASS_HUGE
 	storage_slots = 3
 	can_hold = list() //Nada. Once you take the stuff out it doesn't fit back in.
-	foldable = null
+	//foldable = null
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:42]: ?

File: code/game/objects/items/storage/marine_boxes.dm [hunk 4]
```diff
@@ -1,47 +1,78 @@
...
 	name = "\improper M92 grenade launcher case"
 	desc = "A large case containing a heavy-duty multi-shot grenade launcher, the Armat Systems M92. Drag this sprite into you to open it up!\nNOTE: You cannot put items back inside this case."
-	icon = 'icons/Marine/marine-weapons.dmi'
 	icon_state = "grenade_case"
 	w_class = WEIGHT_CLASS_HUGE
 	storage_slots = 2
 	slowdown = 1
 	can_hold = list() //Nada. Once you take the stuff out it doesn't fit back in.
-	foldable = null
+	//foldable = null
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:58]: ?

File: code/game/objects/items/storage/marine_boxes.dm [hunk 5]
```diff
@@ -1,47 +1,78 @@
...
 	name = "\improper M5 RPG crate"
 	desc = "A large case containing a heavy-caliber antitank missile launcher and missiles. Drag this sprite into you to open it up!\nNOTE: You cannot put items back inside this case."
-	icon = 'icons/Marine/marine-weapons.dmi'
 	icon_state = "rocket_case"
 	w_class = WEIGHT_CLASS_HUGE
 	storage_slots = 6
 	slowdown = 1
 	can_hold = list() //Nada. Once you take the stuff out it doesn't fit back in.
-	foldable = null
+	//foldable = null
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:73]: ? bruh why you doing this, I'm not going to comment on every single one.

File: code/modules/projectiles/guns/sentries.dm [hunk 6]
```diff
@@ -49,15 +48,15 @@
...
 	. = ..()
 	new /obj/item/weapon/gun/sentry/big_sentry(src)
 	new /obj/item/ammo_magazine/sentry(src)
 
-/obj/item/storage/box/sentry/update_icon_state()
+/obj/item/storage/box/crate/sentry/update_icon_state()
 	icon_state = initial(icon_state)
 	if(!length(contents))
-		icon_state += "_e"
+		icon_state = "empty_case"
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/sentries.dm:59]: why does this still exist?

File: code/modules/projectiles/guns/sentries.dm [hunk 7]
```diff
@@ -199,24 +198,28 @@
...
+/obj/item/storage/box/crate/minisentry/Initialize(mapload, ...)
 	. = ..()
 	new /obj/item/weapon/gun/sentry/mini(src)
 	new /obj/item/ammo_magazine/minisentry(src)
 	new /obj/item/ammo_magazine/minisentry(src)
 
+/obj/item/storage/box/crate/minisentry/update_icon_state()
+	icon_state = initial(icon_state)
+	if(!length(contents))
+		icon_state = "empty_case"
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/sentries.dm:221]: Why does this exist?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/marine_boxes.dm (hunk 1 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:29]: It's to change the message when it's folded from "You fold [src] flat." to "You broken [src] into a plank."
I'm very new to coding MD so don't know how to do this better. Do I use ..() or something?

File: code/game/objects/items/storage/marine_boxes.dm (hunk 2 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:34]: Gotcha. So like if(!var/is_empty) kind of deal?
FYI I copied this from the line 377 to 380 from code/game/objects/items/storage/boxes.dm

File: code/game/objects/items/storage/marine_boxes.dm (hunk 3 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:42]: I removed them to ensure they produce a plank of wood when folded. I'll get rid of the comment.

File: code/game/objects/items/storage/marine_boxes.dm (hunk 4 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:58]: I removed them to ensure they produce a plank of wood when folded. I'll get rid of the comment

File: code/game/objects/items/storage/marine_boxes.dm (hunk 5 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:73]: I removed them to ensure they produce a plank of wood when folded. I'll get rid of the comment

File: code/modules/projectiles/guns/sentries.dm (hunk 6 above)
Comment (Sanuki357) [reply code/modules/projectiles/guns/sentries.dm:59]: Forgot to get rid of them. I'll remove them.

File: code/modules/projectiles/guns/sentries.dm (hunk 7 above)
Comment (Sanuki357) [reply code/modules/projectiles/guns/sentries.dm:221]: Forgot to get rid of them. I'll remove them.

File: code/game/objects/items/storage/marine_boxes.dm (hunk 2 above)
Comment (Sanuki357) [reply code/game/objects/items/storage/marine_boxes.dm:34]: Actually should it be like this?
```
if(!length(contents))
		icon_state = "empty_case"
	else
		icon_state = initial(icon_state)
```

File: code/game/objects/items/storage/marine_boxes.dm (hunk 1 above)
Comment (Lumipharon) [reply code/game/objects/items/storage/marine_boxes.dm:29]: either make the parent message more generic "you break down the [src]" or something, or add like a break down quote var

File: code/game/objects/items/storage/marine_boxes.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/items/storage/marine_boxes.dm:34]: that works as well

File: code/game/objects/items/storage/marine_boxes.dm [hunk 8]
```diff
@@ -1,47 +1,50 @@
-/obj/item/storage/box/heavy_armor
+/obj/item/storage/box/crate
+	name = "crate"
+	desc = "It's just an ordinary wooden crate."
+	icon = 'icons/obj/items/storage/storage.dmi'
+	icon_state = "case"
+	foldable = /obj/item/stack/sheet/wood
+
+/obj/item/storage/box/crate/update_icon_state()
+	icon_state = length(contents) ? initial(icon_state) : "empty_case"
```
Comment (Lumipharon) [inline code/game/objects/items/storage/marine_boxes.dm:9]:
```suggestion
	icon_state = length(contents) ? initial(icon_state) : "empty_case"
	
```

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**